crates/db_views_actor/src/community_view.rs

```diff
@@ -173,22 +173,19 @@ impl<'a> CommunityQuery<'a> {
     if !self.is_mod_or_admin.unwrap_or(true) {
       query = query
         .filter(community::removed.eq(false))
-        .filter(community::deleted.eq(false));
+        .filter(community::deleted.eq(false))
+        .filter(
+          community::hidden
+            .eq(false)
+            .or(community_follower::person_id.eq(person_id_join)),
+        );
     }
 
     match self.sort.unwrap_or(SortType::Hot) {
       SortType::New => query = query.order_by(community::published.desc()),
       SortType::TopAll => query = query.order_by(community_aggregates::subscribers.desc()),
       SortType::TopMonth => query = query.order_by(community_aggregates::users_active_month.desc()),
-      SortType::Hot => {
-        query = query.order_by(community_aggregates::hot_rank.desc());
-        // Don't show hidden communities in Hot (trending)
-        query = query.filter(
-          community::hidden
-            .eq(false)
-            .or(community_follower::person_id.eq(person_id_join)),
-        );
-      }
+      SortType::Hot => query = query.order_by(community_aggregates::hot_rank.desc()),
       // Covers all other sorts
       _ => query = query.order_by(community_aggregates::users_active_month.desc()),
     };
```

Comment:
This function is only used for `/communities`? To be honest I dont understand why it needs this kind of ranking anyway, its not even exposed in lemmy-ui. Though I suppose its also used for the "trending communities" in sidebar.